kv/bucket_test.go

```diff
@@ -17,6 +22,75 @@ func TestInmemBucketService(t *testing.T) {
 	influxdbtesting.BucketService(initInmemBucketService, t)
 }
 
+func TestSystemBucketDeletion(t *testing.T) {
```

Comment:
Is there a reason to have these tests in here, and not in `testing/bucket_service.go`? I'm not super worried about functionality diverging between different kv implementations, but it's a little weird that these only run against bolt.

Reply:
the http bucket service doesn't allow for creating buckets with a type of system and it also uses that test group.

Reply:
I suppose the question is, should the http and kv levels be running the same tests?

Reply:
`http/bucket_service.go` is different from `testing/bucket_service.go`. The former is only through the http API, the latter is unit testing of public service functions and runs against bolt, inmem and etcd.

Reply:
I think initially we have the same tests in http level to save some code. But recently, it seems http level has different behaviors, so i think it's not that necessary now